### PR TITLE
Activiti/Activiti#3460-Added CloudApplicationDeployedProducer

### PR DESCRIPTION
**Pull request by Giovanni007**
Application deloyed events are not triggered when rb starts. This PR try to fix the issue

### CODE REVIEW

**Review by andrea-ligios — APPROVED** [review]
LGTM

File: activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/test/java/org/activiti/cloud/services/events/listeners/CloudApplicationDeployedProducerTest.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+    @BeforeEach
+    public void setUp() {
+        initMocks(this);
+        when(producer.auditProducer()).thenReturn(auditProducer);
+        when(runtimeBundleMessageBuilderFactory.create()).thenReturn(messageBuilderAppenderChain);
+    }
+
+    @Test
+    public void shouldSendMessageWithDeployedApplicationWhenWebApplicationTypeIsServlet() {
```
Comment (erdemedeiros) [inline activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/test/java/org/activiti/cloud/services/events/listeners/CloudApplicationDeployedProducerTest.java]: The suffix `WhenWebApplicationTypeIsServlet` seems wrong here as we are not checking the web application type in the code. I think can rename it to `should_sendMessageWithDeployedApplication`.

File: activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/test/java/org/activiti/cloud/services/events/listeners/CloudApplicationDeployedProducerTest.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class CloudApplicationDeployedProducerTest {
```
Comment (erdemedeiros) [inline activiti-cloud-runtime-bundle-service/activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/test/java/org/activiti/cloud/services/events/listeners/CloudApplicationDeployedProducerTest.java:51]: You also need to add an integration test covering the producer, similar to [this one](https://github.com/Activiti/activiti-cloud/blob/develop/activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java#L161).

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Good catch! It's only missing the related integration test.

**Review by erdemedeiros — APPROVED** [review]
LGTM

**Pull request merged**